src/redux/App/reducer.js

```diff
@@ -13,53 +14,58 @@ const initState = new Map({
   current: preKeys,
   walletAddresses: [],
   lastUsedAddress: '',
-  updatingBalances: false,
-  syncProgress: 0,
+  syncPercent: 0,
+  syncBlockNum: 0,
+  syncBlockTime: 0,
   initSyncing: false,
 });
 
 export default function appReducer(state = initState, action) {
   switch (action.type) {
-    case actions.ADD_WALLET_ADDRESS: {
-      const addresses = state.get('walletAddresses');
-      addresses.push({ address: action.value, qtum: 0 });
-      return state.set('walletAddresses', addresses);
+    case actions.TOGGLE_ALL: {
+      if (state.get('view') !== action.view || action.height !== state.height) {
+        const height = action.height ? action.height : state.height;
+        return state
+          .set('collapsed', action.collapsed)
+          .set('view', action.view)
+          .set('height', height);
+      }
+      return state;
     }
     case actions.SET_LAST_USED_ADDRESS: {
       return state.set('lastUsedAddress', action.address);
     }
-    case actions.LIST_UNSPENT_RETURN: {
+    case actions.SYNC_INFO_RETURN: {
       if (action.error) {
-        console.error('Error fetching listunspent');
-        return state;
+        return state.set('syncInfoError', action.error);
       }
 
-      let newAddresses = action.value.addresses;
+      // Process address balances to decimals
+      let newAddresses = [];
+      _.each(action.syncInfo.addressBalances, (addressObj) => {
+        newAddresses.push({
+          address: addressObj.address,
+          qtum: satoshiToDecimal(addressObj.qtum),
+          bot: satoshiToDecimal(addressObj.bot),
+        });
+      });
 
-      // Update the bot balance of the existing addresses or add in existing addresses but with 0 qtum balance.
-      // This is for good UX so the user won't wonder why their old address which they did a sendtoaddress
-      // is no longer on the list of addresses.
-      const existingAddresses = state.get('walletAddresses');
-      _.each(existingAddresses, (addressObj) => {
+      // Add old addresses to new list if not found
+      const oldAddresses = state.get('walletAddresses');
```

Comment:
Is it because after usage, the address balance becomes 0 and it wont be included in listunspent?

Reply:
correct. this should be solved later with the `accounts` db when we add that to keep track of all used addresses.